src/Microsoft.AspNetCore.ResponseCaching/ResponseCache/CachedVaryBy.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.ResponseCaching
+{
+    public class CachedVaryBy : IResponseCacheEntry
```

Comment:
The RFC defines the `Vary` header as a list of other header values to vary by, and this is what the default caches of Apache, Django, RoR support but there are extensions such as node-cache that provide support for vary by params on node. ASP.NET supported a lot more like headers, params, content-encodings and a custom string.